src/detect.c

```diff
@@ -1378,21 +1378,50 @@ static void DetectRunTx(ThreadVars *tv,
 
         /* merge 'state' rules from the regular prefilter */
         uint32_t x = array_idx;
+        uint32_t j = x;
         for (uint32_t i = 0; i < det_ctx->match_array_cnt; i++) {
             const Signature *s = det_ctx->match_array[i];
             if (s->app_inspect != NULL) {
-                const SigIntId id = s->num;
-                det_ctx->tx_candidates[array_idx].s = s;
-                det_ctx->tx_candidates[array_idx].id = id;
-                det_ctx->tx_candidates[array_idx].flags = NULL;
-                det_ctx->tx_candidates[array_idx].stream_reset = 0;
                 array_idx++;
-
-                SCLogDebug("%p/%"PRIu64" rule %u (%u) added from 'match' list",
-                        tx.tx_ptr, tx.tx_id, s->id, id);
             }
         }
-        do_sort = (array_idx > x); // sort if match added anything
+        uint32_t k = array_idx;
```

Comment:
can you move this into a `static inline` helper func? Code is getting too large here

Reply:
ok